Makefile.core.mk

```diff
@@ -78,37 +78,29 @@ clean-proto:
 
 # is all of this needed or is this cruft?
 
-# get imported protos to $GOPATH
 get_dep_proto:
-	GO111MODULE=off go get k8s.io/api/core/v1 k8s.io/api/autoscaling/v2beta1 k8s.io/apimachinery/pkg/apis/meta/v1/ github.com/gogo/protobuf/...
+	GO111MODULE=off GOPATH=/tmp go get k8s.io/api/core/v1 k8s.io/api/autoscaling/v2beta1 k8s.io/apimachinery/pkg/apis/meta/v1/ github.com/gogo/protobuf/...
```

Comment:
interesting solutoin, although these deps are pulled automatically by for example `make download`or are already likely in the image .  Protobuf itself is in the expected systemwide location.

Reply:
The k8s deps are not available in the common image now. Actually pulling in those deps to /tmp using go get does not require changes to existing proto import paths.

Reply:
The common image does include parts of `gogo/protobuf`: https://github.com/istio/tools/blob/master/docker/build-tools/Dockerfile#L46-L49

if parts of `gogo/protbuf` are not present, perhaps you could improve thecommon-tools image to include what is needed?  We really want to pin any dependencies needed using either go.mod or build-tools.

The dependencies are in `go.mod` referenced as `k8s.io/api` and `k8s.io/apimachinery`.  Do you mean these are not pulled into the go package cache when using the repository?  Or just that using `-I` is not desireable with `protoc`?

Reply:
the protoc include path does not handle go package cache properly, that's why I have to pull those deps to temp directory